### DISCUSSION TITLE
Disabling a LSP capability

### DISCUSSION THREAD

**Discussion by LordMZTE** [neovim/neovim#24058] [Q&A]
In previous Neovim versions, it was possible to disable a LSP capbility by setting `client.server_capabilities.documentFormattingProvider = false` in an `on_attach` callback. In a recent update, Neovim has added support for dynamic registration of capabilities and this no longer has any effect, causing `vim.lsp.buf.format` to use the LSP formatter anyways.

**Comment by LordMZTE** [comment]
I found a workaround: I can disable the formatter by using BOTH the old solution and, when starting the client, adding this to the capabilities: `{ textDocument = { formatting = { dynamicRegistration = false } } }`.

Not exactly a good solution.

**Comment by justinmk** [comment]
See also https://github.com/neovim/neovim/issues/24054

**Comment by steventhorne** [comment]
This was pretty necessary for me to get angularls and ts_ls working at the same time. Angularls would get registered first and set a variable that the ts_ls would check against during the on_attach and decide if it should disable the rename and references provider. (Those conflict and cause multiple references to display and asking to rename multiple times) Is there a better way to handle that now?
